var/spack/repos/builtin/packages/py-types-pyyaml/package.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyTypesPyyaml(PythonPackage):
+    """Typing stubs for PyYAML."""
+
+    homepage = "https://github.com/python/typeshed"
+    url      = "https://files.pythonhosted.org/packages/3c/96/22a02c905aec42d8f7f7c48b7a00d68019edd4768e3fb94b754966d07870/types-PyYAML-5.4.6.tar.gz"
```

Comment:
Is there any reason to set the url to the static url with the checksum?
To avoid adding a url specific to a version, and letting spack to resolve the url for different versions, it would be possible to use one of
- https://pypi.io/packages/source/t/types-pyyaml/types-PyYAML-5.4.6.tar.gz
- https://pypi.org/packages/source/t/types-pyyaml/types-PyYAML-5.4.6.tar.gz
- https://files.pythonhosted.org/packages/source/t/types-pyyaml/types-PyYAML-5.4.6.tar.gz

In https://warehouse.readthedocs.io/api-reference/integration-guide.html#querying-pypi-for-package-urls they suggest to use json, but I think it's not possible with spack (but in the upstream release it's possible to use the `pypi` keyword).

Reply:
Addressed in the next PR.